File: src/main/java/com/gitlab/lae/intellij/actions/tree/ReloadAction.java
package com.gitlab.lae.intellij.actions.tree;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.application.ApplicationManager;
import com.intellij.openapi.project.DumbAware;

public final class ReloadAction extends AnAction implements DumbAware {

    @Override
    public void actionPerformed(AnActionEvent e) {
        ApplicationManager
                .getApplication()
                .getComponent(AppComponent.class)
                .reload(e.getActionManager());
    }

}

File: src/main/kotlin/com/gitlab/lae/intellij/actions/tree/ReloadAction.kt
package com.gitlab.lae.intellij.actions.tree

import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.application.ApplicationManager
import com.intellij.openapi.project.DumbAware

class ReloadAction : AnAction(), DumbAware {

    override fun actionPerformed(e: AnActionEvent) {
        ApplicationManager
                .getApplication()
                .getComponent(AppComponent::class.java)
                .reload()
    }

}
